File: rapid-photo-upload/backend/src/main/java/com/rapidphotoupload/api/controllers/AuthController.java
package com.rapidphotoupload.api.controllers;

import com.rapidphotoupload.api.dto.*;
import com.rapidphotoupload.application.commands.CommandResult;
import com.rapidphotoupload.application.commands.RefreshTokenCommand;
import com.rapidphotoupload.domain.aggregates.User;
import com.rapidphotoupload.domain.repositories.UserRepository;
import com.rapidphotoupload.domain.valueobjects.*;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.validation.Valid;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.*;

import java.util.UUID;

/**
 * Controller for authentication endpoints.
 * Note: Login and registration are handled by AWS Cognito on the frontend.
 * This controller provides endpoints for token refresh and getting current user info.
 */
@RestController
@RequestMapping("/api/v1/auth")
public class AuthController {
    
    private static final Logger logger = LoggerFactory.getLogger(AuthController.class);
    
    private final UserRepository userRepository;
    
    public AuthController(UserRepository userRepository) {
        this.userRepository = userRepository;
    }
    
    @PostMapping("/register")
    public ResponseEntity<?> register(@Valid @RequestBody RegisterRequest request, HttpServletRequest httpRequest) {
        // Registration is now handled by AWS Cognito on the frontend
        return ResponseEntity.status(HttpStatus.GONE)
            .body(new ErrorResponse(
                "DEPRECATED",
                "Registration is now handled by AWS Cognito. Please use the frontend registration flow.",
                httpRequest.getRequestURI()
            ));
    }
    
    @PostMapping("/login")
    public ResponseEntity<?> login(@Valid @RequestBody LoginRequest request, HttpServletRequest httpRequest) {
        // Login is now handled by AWS Cognito on the frontend
        return ResponseEntity.status(HttpStatus.GONE)
            .body(new ErrorResponse(
                "DEPRECATED",
                "Login is now handled by AWS Cognito. Please use the frontend login flow.",
                httpRequest.getRequestURI()
            ));
    }
    
    @PostMapping("/refresh")
    public ResponseEntity<?> refresh(@Valid @RequestBody RefreshTokenRequest request, HttpServletRequest httpRequest) {
        // Token refresh is now handled by AWS Cognito on the frontend
        return ResponseEntity.status(HttpStatus.GONE)
            .body(new ErrorResponse(
                "DEPRECATED",
                "Token refresh is now handled by AWS Cognito. Please use the frontend refresh flow.",
                httpRequest.getRequestURI()
            ));
    }
    
    @PostMapping("/logout")
    public ResponseEntity<?> logout(
            @RequestHeader(value = "Authorization", required = false) String authorizationHeader,
            @Valid @RequestBody(required = false) RefreshTokenRequest request,
            HttpServletRequest httpRequest) {
        // Logout is now handled by AWS Cognito on the frontend
        // This endpoint can remain for backward compatibility but does nothing
        return ResponseEntity.ok().build();
    }
    
    @GetMapping("/me")
    public ResponseEntity<?> getCurrentUser(HttpServletRequest httpRequest) {
        try {
            Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
            
            if (authentication == null || !authentication.isAuthenticated()) {
                return ResponseEntity.status(HttpStatus.UNAUTHORIZED)
                    .body(new ErrorResponse(
                        "UNAUTHORIZED",
                        "Not authenticated",
                        httpRequest.getRequestURI()
                    ));
            }
            
            // Extract user info from authentication
            String userId = authentication.getName();
            String email = authentication.getDetails() != null ? authentication.getDetails().toString() : "";
            
            // Try to find user in database by Cognito user ID or email
            User user = null;
            try {
                // First try by email if available
                if (!email.isEmpty()) {
                    user = userRepository.findByEmail(Email.from(email)).orElse(null);
                }
                
                // If not found by email, try by ID (convert Cognito UUID to our UserId)
                if (user == null) {
                    try {
                        UserId userIdObj = UserId.from(UUID.fromString(userId));
                        user = userRepository.findById(userIdObj).orElse(null);
                    } catch (IllegalArgumentException e) {
                        // userId is not a valid UUID, skip
                    }
                }
            } catch (Exception e) {
                logger.warn("Failed to lookup user in database: {}", e.getMessage());
            }
            
            // Return user info (from database if found, otherwise from token)
            if (user != null) {
                AuthResponse.UserInfo userInfo = new AuthResponse.UserInfo(
                    user.getId().getValue(),
                    user.getUsername().getValue(),
                    user.getEmail().getValue()
                );
                return ResponseEntity.ok(userInfo);
            } else {
                // User not in database yet, return info from token
                // Generate a UUID from the Cognito sub (userId) if possible, otherwise create a new one
                UUID userUuid;
                try {
                    userUuid = UUID.fromString(userId);
                } catch (IllegalArgumentException e) {
                    // If userId is not a valid UUID, generate a deterministic one from the userId string
                    userUuid = UUID.nameUUIDFromBytes(userId.getBytes());
                }
                
                AuthResponse.UserInfo userInfo = new AuthResponse.UserInfo(
                    userUuid,
                    email.split("@")[0], // Use email prefix as username
                    email.isEmpty() ? userId : email
                );
                return ResponseEntity.ok(userInfo);
            }
        } catch (Exception e) {
            logger.error("Failed to get current user", e);
            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
                .body(new ErrorResponse(
                    "INTERNAL_SERVER_ERROR",
                    "An unexpected error occurred",
                    httpRequest.getRequestURI()
                ));
        }
    }
}

File: smart-scheduler/backend/Controllers/AuthController.cs
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using System.Security.Claims;
using SmartScheduler.Commands;
using SmartScheduler.DTOs;
using MediatR;

namespace SmartScheduler.Controllers;

[ApiController]
[Route("api/auth")]
public class AuthController : ControllerBase
{
    private readonly IMediator _mediator;
    private readonly ILogger<AuthController> _logger;

    public AuthController(IMediator mediator, ILogger<AuthController> logger)
    {
        _mediator = mediator;
        _logger = logger;
    }

    [HttpPost("register")]
    [AllowAnonymous]
    [ProducesResponseType(StatusCodes.Status410Gone)]
    public ActionResult Register([FromBody] RegisterRequest request)
    {
        // Registration is now handled by AWS Cognito on the frontend
        return StatusCode(410, new {
            message = "Registration is now handled by AWS Cognito. Please use the frontend registration flow."
        });
    }

    [HttpPost("login")]
    [AllowAnonymous]
    [ProducesResponseType(StatusCodes.Status410Gone)]
    public ActionResult Login([FromBody] LoginRequest request)
    {
        // Login is now handled by AWS Cognito on the frontend
        return StatusCode(410, new {
            message = "Login is now handled by AWS Cognito. Please use the frontend login flow."
        });
    }

    [HttpGet("me")]
    [AllowAnonymous] // Allow public access - no authentication required
    [ProducesResponseType(typeof(object), StatusCodes.Status200OK)]
    public ActionResult GetCurrentUser()
    {
        var requestId = HttpContext.Items["RequestId"]?.ToString() ?? "unknown";
        _logger.LogInformation("[RequestId: {RequestId}] GetCurrentUser called, Authenticated={Authenticated}",
            requestId, User?.Identity?.IsAuthenticated ?? false);
        
        try
        {
            var userId = User.FindFirst(ClaimTypes.NameIdentifier)?.Value 
                ?? User.FindFirst("sub")?.Value 
                ?? User.Identity?.Name;
            var email = User.FindFirst(ClaimTypes.Email)?.Value 
                ?? User.FindFirst("email")?.Value;
            var username = User.FindFirst(ClaimTypes.Name)?.Value 
                ?? User.FindFirst("cognito:username")?.Value
                ?? (email != null ? email.Split('@')[0] : null);

            _logger.LogInformation("[RequestId: {RequestId}] GetCurrentUser completed, userId={UserId}, email={Email}",
                requestId, userId, email);

            return Ok(new
            {
                id = userId,
                email = email,
                username = username
            });
        }
        catch (Exception ex)
        {
            _logger.LogError(ex, "[RequestId: {RequestId}] Error in GetCurrentUser", requestId);
            return StatusCode(500, new { message = "An error occurred while getting user info", error = ex.Message });
        }
    }
}

